### PR TITLE
Backend [Integration Test] people.go handlers GetPeopleBySearch, GetListedPeople, & GetPersonByUuid

### PR DESCRIPTION
**Pull request by AbdulWahab3181**
## Describe your changes

Added Unit test for people handler to test GetPeopleBySearch, GetListedPeople and GetPersonByUuid methods

## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes/issues/1518

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by aliraza556** [comment]
Overall, the unit test code provided meets the specified requirements for testing the people handlers. Each required scenario is covered by a corresponding test case, ensuring comprehensive test coverage for the functionality described.

File: handlers/people_test.go [hunk 1]
```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
...
+	t.Run("should return users that match the search text", func(t *testing.T) {
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetPeopleBySearch)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("search", "John")
```
Comment (elraphty) [inline handlers/people_test.go:275]: @AbdulWahab3181 the search should be in the URL query, not param.

File: handlers/people_test.go [hunk 2]
```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
...
+	})
+
+	t.Run("should return an empty search result when no user matches the search text", func(t *testing.T) {
+		mockDb.ExpectedCalls = nil
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetPeopleBySearch)
+		expectedPeople := []db.Person{}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("search", "user not matched")
```
Comment (elraphty) [inline handlers/people_test.go:297]: @AbdulWahab3181 the search should be in the URL query, not param.

File: handlers/people_test.go [hunk 3]
```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
...
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
+		rctx.URLParams.Add("limit", "10")
```
Comment (elraphty) [inline handlers/people_test.go:327]: @AbdulWahab3181 the limit and page should be in the URL query, not param.

File: handlers/people_test.go [hunk 4]
```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
...
+		mockDb.ExpectedCalls = nil
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+			{ID: 2, Uuid: "uuid2", OwnerPubKey: "pubkey2", OwnerAlias: "John Smith"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
```
Comment (elraphty) [inline handlers/people_test.go:352]: @AbdulWahab3181 the limit, page, and search should be in the URL query, not param.

File: handlers/people_test.go [hunk 5]
```diff
@@ -258,3 +258,191 @@ func TestDeletePerson(t *testing.T) {
...
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(pHandler.GetListedPeople)
+		expectedPeople := []db.Person{
+			{ID: 1, Uuid: "uuid1", OwnerPubKey: "pubkey1", OwnerAlias: "John Doe"},
+		}
+
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("page", "1")
+		rctx.URLParams.Add("limit", "10")
+		rctx.URLParams.Add("languages", "typescript")
```
Comment (elraphty) [inline handlers/people_test.go:380]: @AbdulWahab3181 the page, limit, and languages should be in the URL query, not the param.

File: handlers/people_test.go (hunk 1 above)
Comment (AbdulWahab3181) [reply handlers/people_test.go]: @elraphty  Addressed

File: handlers/people_test.go (hunk 2 above)
Comment (AbdulWahab3181) [reply handlers/people_test.go]: Addressed

File: handlers/people_test.go (hunk 4 above)
Comment (AbdulWahab3181) [reply handlers/people_test.go]: Addressed

File: handlers/people_test.go (hunk 5 above)
Comment (AbdulWahab3181) [reply handlers/people_test.go]: Addressed

**Pull request merged**